### PR TITLE
updated go to 1.17

### PR DESCRIPTION
**Pull request by jbyers19**
Signed-off-by: Jared Byers <j.byers@f5.com>

**Description**
*  Updated go version in `go.mod` to `1.17`. 
*  Added a `replace` for the OTel modules to prevent version mismatch with other components.
*  Bumped `meshkit` to `v0.5.12` instead of `v0.5.13` to avoid dependency issues introduced in `v0.5.13`.
*  Bumped SMP to `v0.3.4`.
*  All other changes were generated by `go mod tidy`.

Ran `go mod verify` and `go vet ./...` locally to test changes.

**Edit:**
Had to make a couple more changes to fix lint failures:
1. Updated `golangci-lint` to ~~v1.45~~ **_latest_** in the workflow. 
2. Switched from `strings.Title` to `cases.Title`.
Tested the change locally to ensure the output would be the same:
```
mesh_type := "NGINX_SERVICE_MESH"
strings_name := strings.Title(strings.ToLower(strings.ReplaceAll(mesh_type, "_", " ")))
cases_name := cases.Title(language.Und).String(strings.ReplaceAll(mesh_type, "_", " "))

Output:
strings: Nginx Service Mesh
cases: Nginx Service Mesh
```

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@alphaX86 given you recent focus on golanglint-ci, will you have a look at the failed check here?

**Comment by jbyers19** [comment]
@alphaX86, I bumped golangci-lint to the latest version (v1.45) and this fixed most of the errors. However, there's still 1 error in the GitHub action. I ran the same command locally with the same version of `golangci-lint` and am not seeing this error. Is there a caching issue perhaps?

Here's the command I ran locally. The only output is this warning, but no errors.
```
$ golangci-lint run --out-format=github-actions --exclude G306
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
```

**Comment by jbyers19** [comment]
> @alphaX86 given you recent focus on golanglint-ci, will you have a look at the failed check here?

Thanks, @leecalcote. I figured how to fix the lint issues. See the updated description for details.

**Comment by alphaX86** [comment]
> **Update:** fixed the lint error via code change. Still don't know why the error didn't show up locally.
> 
> @alphaX86, I bumped golangci-lint to the latest version (v1.45) and this fixed most of the errors. However, there's still 1 error in the GitHub action. I ran the same command locally with the same version of `golangci-lint` and am not seeing this error. Is there a caching issue perhaps?
> 
> Here's the command I ran locally. The only output is this warning, but no errors.
> 
> ```
> $ golangci-lint run --out-format=github-actions --exclude G306
> WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
> ```

It might be due to the old config present in repo. Seems you've fixed all issues 😅 sorry I was in rest at that time. Anyway 👍

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
         uses: golangci/golangci-lint-action@v2
         with:
           # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
-          version: v1.32
+          version: v1.45
```
Comment (alphaX86) [inline .github/workflows/ci.yml:25]: Better we may use tag 'latest' instead of specific version number...

**Review by alphaX86 — COMMENTED** [review]
```
	  ​      - ​uses​: ​actions/checkout@v2 
		    - uses: actions/setup-go@v3
		    with:
		    go-version: '1.17'
		    ​      - ​name​: ​golangci-lint 
		    ​        ​uses​: ​golangci/golangci-lint-action@v2 
		    ​        ​with​: 
		    ​          ​#​ Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version. 
		    ​          ​version​: ​v1.45
```

OK, I'm just giving rough draft... here in this step we might need to add additional step to setup go1. 17...

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ jobs:
         uses: golangci/golangci-lint-action@v2
```
Comment (alphaX86) [inline .github/workflows/ci.yml:22]:
```suggestion
	- uses: action/setup-go@v3
	  with:
	    go-version: '1.17'
	    check-latest: 'true'
        uses: golangci/golangci-lint-action@v2
```

**Comment by jbyers19** [comment]
Thanks, @alphaX86. I updated the workflow with your suggestions.

**Review by alphaX86 — APPROVED** [review]
LGTM 👍🔥

**Review by leecalcote — APPROVED** [review]
Looking good. 👌

**Comment by jbyers19** [comment]
@leecalcote, can this be merged now?

**Review by MUzairS15 — APPROVED** [review]

**Pull request merged**